docs/databases/postgresql/how-to-deploy-ha-postgresql-cluster.md

```diff
@@ -0,0 +1,406 @@
+
+{: .file}
+/etc/systemd/system/patroni.service
+:   ~~~
+    [Unit]
+    Description=Runners to orchestrate a high-availability PostgreSQL
+    After=syslog.target network.target
+
+    [Service]
+    Type=simple
+
+    User=postgres
+    Group=postgres
+
+    ExecStart=/usr/local/bin/patroni /etc/patroni.yml
+
+    KillMode=process
+
+    TimeoutSec=30
+
+    Restart=no
+
+    [Install]
+    WantedBy=multi-user.targ
+    ~~~
+
+{: .note}
+>
+> If `patroni` is installed in a location other than `/usr/local/bin/patroni` on your machine, update the above file accordingly.
+
+To start Patroni (and thereby Postgres), execute the following command:
+
+    sudo systemctl start patroni
+
+To check the status of Patroni, run the following command:
+
+    sudo systemctl status patroni
+
+If everything is setup correctly, this should show an output similar to the following:
+
+    ● patroni.service - Runners to orchestrate a high-availability PostgreSQL
+    Loaded: loaded (/etc/systemd/system/patroni.service; enabled; vendor preset: enabled)
+    Active: active (running) since Thu 2017-07-29 16:49:18 UTC; 8min ago
+    Main PID: 13097 (patroni)
+
+    .
+    .
+    .
+
+    ... INFO: Lock owner: postgresql0; I am postgresql0
+    ... INFO: no action.  i am the leader with the lock
+
+This is what you should see when starting up the first node (leader). When starting up subsequent nodes, you should see a log that contains:
+
+    INFO: no action.  i am a secondary and i am following a leader
+    Lock owner: postgresql0; I am postgresql2
+
+Doing this on all the three Linodes will create a highly available Postgres cluster with one master and two slaves.
+
+### Configuring HAProxy
+
+With the Postgres cluster set up, we need a way to connect to the master regardless of which of the servers in the cluster is the master. This is where HAProxy comes in. All Postgres clients (your applications, `psql`, etc.) will connect to HAProxy which will make sure you get connected to the master in the cluster.
+
+On the machine that has HAProxy installed, edit the configuration file at `/etc/haproxy/haproxy.cfg` to contain the following:
+
+{: .file}
+/etc/haproxy/haproxy.cfg
+:   ~~~
+    global
+        maxconn 100
+
+    defaults
+        log global
+        mode tcp
+        retries 2
+        timeout client 30m
+        timeout connect 4s
+        timeout server 30m
+        timeout check 5s
+
+    listen stats
+        mode http
+        bind *:7000
+        stats enable
+        stats uri /
+
+    listen postgres
+        bind *:5000
+        option httpchk
+        http-check expect status 200
+        default-server inter 3s fall 3 rise 2 on-marked-down shutdown-sessions
+        server postgresql_192.0.2.11_5432 192.0.2.11:5432 maxconn 100 check port 8008
+        server postgresql_192.0.2.12_5432 192.0.2.12:5432 maxconn 100 check port 8008
+        server postgresql_192.0.2.13_5432 192.0.2.13:5432 maxconn 100 check port 8008
+    ~~~
+
```

Comment:
Neat feature for checking syntax: `/usr/sbin/haproxy -c -V -f /etc/haproxy/haproxy.cfg`